### PR TITLE
Added functionality to search multiple udi and dois

### PR DESCRIPTION
**Pull request by praneethpr**

### CODE REVIEW

File: src/Util/Search.php [hunk 1]
```diff
@@ -890,9 +893,12 @@ private function doesUdiExistInQueryTerm(string $queryTerm, Query\BoolQuery $fie
         if (preg_match_all($udiRegEx, $queryTerm, $matches)) {
```
Comment (mickel1138) [inline src/Util/Search.php:893]: The regex needs the `g` flag. For global, to match all matches.

File: src/Util/Search.php [hunk 2]
```diff
@@ -890,9 +893,12 @@ private function doesUdiExistInQueryTerm(string $queryTerm, Query\BoolQuery $fie
...
             trim(preg_replace($udiRegEx, '', $queryTerm));
             $queryTerm = $matches[1][0];
-            // Replacing the 11th position to ":"
-            $queryTerm = substr_replace($queryTerm, ':', 11, 1);
-            $fieldsBoolQuery->addShould($this->getUdiQuery($queryTerm));
+            foreach ($matches[1] as $udi) {
+                $queryTerm = $queryTerm . ' ' . $udi;
+                // Replacing the 11th position to ":"
+                $udiQuery = substr_replace($udi, ':', 11, 1);
```
Comment (mickel1138) [inline src/Util/Search.php:899]: I would use a `preg_replace` here, to find the pattern to fix, and fix it globally.

File: src/Util/Search.php [hunk 3]
```diff
@@ -890,9 +893,12 @@ private function doesUdiExistInQueryTerm(string $queryTerm, Query\BoolQuery $fie
         if (preg_match_all($udiRegEx, $queryTerm, $matches)) {
             trim(preg_replace($udiRegEx, '', $queryTerm));
             $queryTerm = $matches[1][0];
-            // Replacing the 11th position to ":"
-            $queryTerm = substr_replace($queryTerm, ':', 11, 1);
-            $fieldsBoolQuery->addShould($this->getUdiQuery($queryTerm));
+            foreach ($matches[1] as $udi) {
+                $queryTerm = $queryTerm . ' ' . $udi;
```
Comment (mickel1138) [inline src/Util/Search.php:897]: Wouldn't this add more udi's to the queryTerm?

**Review by mickel1138 — DISMISSED** [review]
I approve of these changes.

**Review by fightingtexasaggie — APPROVED** [review]
That turned out suprisingly simple! Sweet.

**Review by mickel1138 — APPROVED** [review]

**Pull request merged**